File: LocationVoiture/src/main/java/model/Loueur.java
package model;


import java.util.List;

import javax.persistence.Entity;
import javax.persistence.JoinTable;
import javax.persistence.OneToMany;


import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Loueur extends Utilisateur {

	@JsonView(JsonViews.Common.class)
	@OneToMany(mappedBy = "loueur")
	private List<Annonce> annonce;

	public Loueur() {
	}

	
	
	

	public Loueur(String password, String login, String nom, String prenom, List<Annonce> annonce) {
		super(password, login, nom, prenom);
		this.annonce = annonce;
	}





	public Loueur(String password, String login, String nom, String prenom, Adresse adresse, List<Annonce> annonce) {
		super(password, login, nom, prenom, adresse);
		this.annonce = annonce;
	}





	public List<Annonce> getAnnonce() {
		return annonce;
	}





	public void setAnnonce(List<Annonce> annonce) {
		this.annonce = annonce;
	}





	

}

File: LocationVoitureAngular/src/app/location/model/loueur.ts
import { Adresse } from './adresse';
import { Compte } from './compte';

export class Loueur extends Compte {
  constructor(
    _id?: number,
    _nom?: string,
    _prenom?: string,
    _email?: string,
    _adresse?: Adresse
  ) {
    super(_id, _nom, _prenom, _email, _adresse);
  }
}
